packages/fast-usdc/src/exos/status-manager.js

```diff
@@ -86,25 +70,71 @@ export const prepareStatusManager = (
     log = makeTracer('Advancer', true),
   } = /** @type {StatusManagerPowers} */ ({}),
 ) => {
-  /** @type {MapStore<PendingTxKey, PendingTx[]>} */
+  /**
+   * Keyed by a tuple of the Noble Forwarding Account and amount.
+   * @type {MapStore<PendingTxKey, PendingTx[]>}
+   */
   const pendingTxs = zone.mapStore('PendingTxs', {
     keyShape: M.string(),
     valueShape: M.arrayOf(PendingTxShape),
   });
 
-  /** @type {SetStore<SeenTxKey>} */
+  /**
+   * Transactions seen *ever* by the contract.
+   *
+   * Note that these are stored in IAVL and grow monotonically. At some point in the future we may want to prune them.
+   *
+   * @type {SetStore<EvmHash>}
+   */
   const seenTxs = zone.setStore('SeenTxs', {
     keyShape: M.string(),
   });
 
+  /**
+   * Transactions that have completed, but are still in vstorage.
+   *
+   * @type {SetStore<EvmHash>}
+   */
+  const storedCompletedTxs = zone.setStore('StoredCompletedTxs', {
+    keyShape: M.string(),
+  });
+
+  /**
+   * @param {EvmHash} txId
+   * @param {CopyRecord} record
+   */
+  const publishTxnRecord = (txId, record) => {
+    const txNode = E(transactionsNode).makeChildNode(txId, {
+      sequence: true, // avoid overwriting other output in the block
+    });
+    void E(txNode).setValue(
+      JSON.stringify(pureDataMarshaller.toCapData(harden(record))),
```

Comment:
`harden(...)` doesn't hurt (much) here, but if it's already a `CopyRecord`, it's already hardened.

Reply:
>harden(...) doesn't hurt (much) here
The marshaller threw without it. 

>if it's already a CopyRecord, it's already hardened

Not according to the structural type of CopyRecord,
```
export type CopyRecord<T extends Passable = any> = Record<string, T>;
```

Reply:
> The marshaller threw without it.

That's a a bug in the caller that claimed something was a `CopyRecord` without `harden`-ing it.

This is actually the cost of adding the `harden(...)` here: it masks such bugs.